usaspending_api/references/views_v2.py

```diff
@@ -0,0 +1,38 @@
+from django.db.models import F, Sum
+from usaspending_api.references.models import Agency
+from usaspending_api.submissions.models import SubmissionAttributes
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+
+class AgencyEndpoint(APIView):
+    """Return an agency name and active fy"""
+    def get(self, request, pk, format=None):
+        """Return the view's queryset."""
+        # get id from url
+        agency_id = int(pk)
+        # get agency cgac code
+        agency = Agency.objects.filter(id=agency_id).first()
+
+        if agency is None:
+            return Response({'results': {}})
+
+        toptier_agency = agency.toptier_agency
+        # get corresponding submissions through cgac code
+        queryset = SubmissionAttributes.objects.all()
+        queryset = queryset.filter(cgac_code=toptier_agency.cgac_code)
+
+        # get the most up to date fy
+        queryset = queryset.order_by('-reporting_fiscal_year')
+        queryset = queryset.annotate(
+            fiscal_year=F('reporting_fiscal_year')
+        )
+        submission = queryset.first()
+        if submission is None:
+            return Response({'results': {}})
```

Comment:
Here, above line 31, I would create a response dict: `response = {}` and return `Response({'results': response})`. And then later where you're setting up the agency name and fiscal year, you can just add that to the response dict: `response['agency_name'] = toptier_agency.name` and etc. and do a similar return as above.

This method allows easier expansion to the response contents as opposed to adding to the return statement on line 38